hazelcast/src/main/java/com/hazelcast/map/impl/MapConfigValidator.java

```
@@ -36,36 +37,53 @@ private MapConfigValidator() {
+    }
+
+    /**
+     * Throws {@link IllegalArgumentException} if the supplied {@link InMemoryFormat} is {@link InMemoryFormat#NATIVE}
+     *
+     * @param inMemoryFormat supplied inMemoryFormat.
+     */
+    private static void checkNotNative(InMemoryFormat inMemoryFormat) {
+        if (NATIVE == inMemoryFormat) {
+            throw new IllegalArgumentException("NATIVE storage format is supported in Hazelcast Enterprise only. "
+                    + "Make sure you have Hazelcast Enterprise JARs on your classpath!");
+        }
     }
 
     @SuppressWarnings("deprecation")
     private static void logIgnoredConfig(MapConfig mapConfig) {
-        if (DEFAULT_MIN_EVICTION_CHECK_MILLIS != mapConfig.getMinEvictionCheckMillis()
-                || DEFAULT_EVICTION_PERCENTAGE != mapConfig.getEvictionPercentage()) {
+        if (mapConfig.getMinEvictionCheckMillis() != DEFAULT_MIN_EVICTION_CHECK_MILLIS
+                || mapConfig.getEvictionPercentage() != DEFAULT_EVICTION_PERCENTAGE) {
+            LOGGER.warning("As of version 3.7, `minEvictionCheckMillis` and `evictionPercentage`"
+                    + " are deprecated due to the eviction mechanism change. New eviction mechanism"
+                    + " uses a probabilistic algorithm based on sampling. Please see documentation for further details");
+        }
+    }
 
-            LOGGER.warning("As of version 3.7, `minEvictionCheckMillis` and `evictionPercentage` "
-                    + "are deprecated due to the eviction mechanism change. New eviction mechanism "
-                    + "uses a probabilistic algorithm based on sampling. Please see documentation for further details");
+    private static void logIgnoredConfig(NearCacheConfig nearCacheConfig, boolean isClient) {
+        if (isClient && nearCacheConfig.isCacheLocalEntries()) {
+            LOGGER.warning("The Near Cache option `cache-local-entries` is not supported in client configurations!");
```

Fail fast.. fail hard. One of our architectural principles